ui/src/app/pages/workspace/workspace-share.spec.tsx

```diff
@@ -236,18 +257,19 @@ describe('WorkspaceShare', () => {
       .first()
       .simulate('click');
     wrapper.update();
+
     const spy = jest.spyOn(workspacesApi(), 'shareWorkspace');
     wrapper.find('[data-test-id="save"]').first().simulate('click');
     expect(spy).toHaveBeenCalledWith(
       tomRiddleDiary.namespace,
       tomRiddleDiary.name,
       {
         workspaceEtag: tomRiddleDiary.etag,
-        items: fp.sortBy('familyName', [
```

Comment:
At test start we have [harry/Owner, hermione/Writer, ron/Reader]
We perform [add luna, remove ron, change hermione].  

Previous logic: must specify all retained users, and any users left off are removed.
So the change request is [harry/Owner, luna/Reader, hermione/Reader].  ron is implictly removed.

New logic: only specify changes, but we need to include removals.
So the change request is [luna/Reader, ron/No-Access, hermione/Reader]